### PR TITLE
IGNITE-19312 Apache Ignite 2.15.0 release notes

### PR DESCRIPTION
**Pull request by alex-plekhanov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: RELEASE_NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,156 @@
...
+
+Java thin-client:
+* Added endpoints discovery feature.
+* Added ClientConfiguration.clusterDiscoveryEnabled configuration property.
+* Added events listening.
+* Added connections balancing (use random open connection for requests instead of default).
+* Added configurable logging.
+* Fixed the pending requests race on close.
+
+Ignite .Net:
```
Comment (ptupitsyn) [inline RELEASE_NOTES.txt:113]:
```suggestion
Ignite .NET:
```

Comment (alex-plekhanov) [reply RELEASE_NOTES.txt]: Fixed, thanks

File: RELEASE_NOTES.txt [hunk 2]
```diff
@@ -1,6 +1,156 @@
...
+* Added ability to set up global heap memory quota and per-query hep memory quota .
+* Added ability to scan inlined keys of indexes without additional requests to data pages.
+* Added native support for SEARCH/SARG operator (this makes possible index scans for IN and some other conditions).
+* Added support for LocalDate/LocalTime types.
+* Added support of index scans on boolean fields.
+* Added EVERY/SOME aggregate functions.
+* Added details to Calcite parser exception.
+* Added new distribution type (optimization to make some joins colocated).
+* Added optimization for COUNT(column) if column is indexed.
+* Added optimization: merge index conditions for the same field .
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:144]: unnecessary space

File: RELEASE_NOTES.txt [hunk 3]
```diff
@@ -1,6 +1,156 @@
...
+ODBC:
+* Fixed handling of single quotes.
+
+SQL:
+* Fixed redundant query results for commented lines.
+* Fixed incorrect type conversion for index scans (for example, when indexed field has string data type and search row has numeric data type).
+* Fixed potential deadlock in discovery thread while updating SQL statistics.
+
+Calcite-based SQL engine:
+* Added ability to set up global heap memory quota and per-query hep memory quota .
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:135]: unnecessary space

File: RELEASE_NOTES.txt [hunk 4]
```diff
@@ -1,6 +1,156 @@
...
+* Added maximum CDC directory size configuration parameter.
+* Added expiry policy support for CDC.
+* Added ability to provide expireTime during CDC conflict resolution.
+* Added LOB and BLOB fields support to CacheJdbcPojoStore.
+* Added missing "--add-opens" for Java 15+ runtime.
+* Added default constructor to PlatformDotNetSessionLockResult to avoid issues with ReflectionFactory on some application servers.
+* Added ability to use configured FailureHandler for segmentation handling.
+* Added authorization of .NET and C++ compute tasks.
+* Added authorization of Ignite Cluster Node stop/start/restart operations.
+* Added authorization of SQL user managemnt operaitons.
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:41]: typo: management

File: RELEASE_NOTES.txt [hunk 5]
```diff
@@ -1,6 +1,156 @@
...
+* Added 'ADMIN_CLUSTER_STATE' security permission (now it's the only required permission to change cluster state).
+* Granting permissions to system compute tasks by their FQN has been replaced with predefined system permissions.
+* Improved rebalance and PME logging (partition ranges disabled).
+* Default metrics exportes is now the JMX exporter.
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:45]: typo: exporter

File: RELEASE_NOTES.txt [hunk 6]
```diff
@@ -1,6 +1,156 @@
...
+* Optimized index rebuilding (disable writing of WAL records for indexes during rebuilding) .
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:46]: unnecessary space

File: RELEASE_NOTES.txt [hunk 7]
```diff
@@ -1,6 +1,156 @@
...
+* Fixed skipping verification of index and metastorage partitions in the snapshot verification process.
+* Fixed criteria skipping after first IN clause for IndexQuery.
+* Fixed WalWritingRate metric for LOG_ONLY and BACKGROUND modes.
+* Updated log4j2 dependency to version 2.20.0.
+* Updated Jetty dependency to 9.4.49.v20220914.
+* Updated Jackson Databind dependency to 2.14.0.
+* Updated postgresql dependency to 42.5.0 .
+* Updated Netty dependency to 4.1.89.Final.
+* Updated Ant dependency to 1.10.13.
+* Updaded Apache Calcite dependency to 1.32.0.
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:89]: typo: Updated

**Review by NSAmelchev — CHANGES_REQUESTED** [review]

File: RELEASE_NOTES.txt [hunk 8]
```diff
@@ -1,6 +1,156 @@
...
+Apache Ignite In-Memory Distributed Database 2.15.0
+-----------------------------------------------------------
+
+(!) WARNINGS:
+* Removed deprecated daemon mode for Ignite nodes.
+* Removed deprecated ignitevisorcmd tool.
+* Removed legacy JMX metric beans (ThreadPoolMXBean, CacheGroupMetricsMXBean, CacheMetricsMXBean, PersistenceMetricsMXBean, DataStorageMetricsMXBean,  DataRegionMetricsMXBean).
+* Removed redundant ignite-spring to ignite-indexing module dependency. If you use SQL in Ignite and get Ignite dependency via maven, check that you have explicit dependency to the one of the SQL engines (ignite-indexing or ignite-calcite).
+* Moved ignite-ssh module to the ignite-extensions.
+* Deprecated lazy flag of SqlFieldsQuery.
```
Comment (NSAmelchev) [inline RELEASE_NOTES.txt:13]: Lets add that default value is now `true`

**Pull request closed without merging**